cli/client/http.go

```diff
@@ -107,6 +107,43 @@ func (c *HTTPClient) RetryPollStep(ctx context.Context, in *api.PollStepRequest,
 	}
 }
 
+func (c *HTTPClient) GetStepLogOutput(ctx context.Context, in *api.StreamOutputRequest, w io.Writer) error {
```

Comment:
This is not the correct approach. We are already writing the logs to a file. Check: https://github.com/harness/lite-engine/blob/main/logstream/filestore/filestore.go#L70

In case of log streaming, read this file, tail the logs and close the stream when step is complete. It will also ensure that if stream log call fails in between, retry will work. Also, there won't be any waiting for step to start or anything like that.

Reply:
Use this utility for tailing the logs from the file: https://pkg.go.dev/github.com/nxadm/tail#section-readme

Let's discuss the approach on monday.